quic/s2n-quic/src/client/providers.rs

```diff
@@ -297,7 +297,7 @@ impl<
     type TLSEndpoint = Tls;
     type TokenFormat = Token;
     type ConnectionLimits = Limits;
-    type Stream = stream::StreamImpl;
```

Comment:
Sort of confused here why we could replace a Stream with a StreamManager.

Reply:
The type wasn't exposed or overridable anywhere. The `Stream` was designed to be swappable but it's much more useful to swap the manager itself.

Either way; the `StreamManager` type isn't overridable either so it's still just an implementation detail.